src/rpdk/core/contract/resource_client.py

```diff
@@ -242,6 +282,38 @@ def assert_time(start_time, end_time, action):
             "Handler %r timed out." % action
         )
 
+    @staticmethod
+    def get_decoded_identifier_list(primary_identifier_path):
+        tuple_identifier = []
+        primary_identifier_list = []
```

Comment:
use [list comprehension](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions) here. What is this method doing? encoding then decoding? seems like no-op